app/platform_stats/rest.py

```diff
@@ -27,3 +32,75 @@ def get_platform_stats():
     stats = format_admin_stats(data)
 
     return jsonify(stats)
+
+
+def validate_date_range_is_within_a_financial_year(start_date, end_date):
+    try:
+        start_date = datetime.strptime(start_date, "%Y-%m-%d").date()
+        end_date = datetime.strptime(end_date, "%Y-%m-%d").date()
+    except ValueError:
+        raise InvalidRequest(message="Input must be a date in the format: YYYY-MM-DD", status_code=400)
+    if end_date < start_date:
+        raise InvalidRequest(message="Start date must be before end date", status_code=400)
+    if 4 <= int(start_date.strftime("%m")) <= 12:
+        year_start, year_end = get_financial_year(year=int(start_date.strftime("%Y")))
+    else:
+        year_start, year_end = get_financial_year(year=int(start_date.strftime("%Y")) - 1)
+    year_start = year_start.date()
+    year_end = year_end.date()
+    if year_start <= start_date <= year_end and year_start <= end_date <= year_end:
+        return True
+    else:
+        raise InvalidRequest(message="Date must be in a single financial year.", status_code=400)
+
+
+@platform_stats_blueprint.route('usage-for-all-services')
+def get_usage_for_all_services():
+    start_date = request.args.get('start_date')
+    end_date = request.args.get('end_date')
+
+    validate_date_range_is_within_a_financial_year(start_date, end_date)
+    start_date = datetime.strptime(start_date, "%Y-%m-%d")
+    end_date = datetime.strptime(end_date, "%Y-%m-%d")
+
+    sms_costs = fetch_sms_billing_for_all_services(start_date, end_date)
+    letter_costs = fetch_letter_costs_for_all_services(start_date, end_date)
+    letter_breakdown = fetch_letter_line_items_for_all_services(start_date, end_date)
+
+    lb_by_service = [
+        (lb.service_id, "{} {} class letters at {}p".format(lb.letters_sent, lb.postage, int(lb.letter_rate * 100)))
+        for lb in letter_breakdown
+    ]
+    combined = {}
+    for s in sms_costs:
+        entry = {
+            "organisation_id": str(s.organisation_id) if s.organisation_id else "",
+            "organisation_name": s.organisation_name or "",
+            "service_id": str(s.service_id),
+            "service_name": s.service_name,
+            "sms_cost": float(s.sms_cost),
+            "sms_fragments": s.chargeable_billable_sms,
+            "letter_cost": 0,
+            "letter_breakdown": ""
+        }
+        combined[str(s.service_id)] = entry
```

Comment:
now that we're no longer jsonifying the dict, we don't need to cast the uuids to strings to use as keys any more